test/mongodb.test.js

```diff
@@ -1406,6 +1406,37 @@ describe('mongodb connector', function() {
     });
   });
 
+  it('findOrCreate should properly support field projection (on create)', function() {
```

Comment:
Can you add a similar test where `fields` is passed in as an array?

Reply:
done